kubernetes/utils/create_from_yaml.py

```diff
@@ -23,51 +23,105 @@
-        # Take care for the case e.g. api_type is "apiextensions.k8s.io"
-        # Only replace the last instance
-        group = "".join(group.rsplit(".k8s.io", 1))
-        fcn_to_call = "{0}{1}Api".format(group.capitalize(),
-                                         version.capitalize())
-        k8s_api = getattr(client, fcn_to_call)(k8s_client)
-        # Replace CamelCased action_type into snake_case
-        kind = yml_object["kind"]
-        kind = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', kind)
-        kind = re.sub('([a-z0-9])([A-Z])', r'\1_\2', kind).lower()
-        # Decide which namespace we are going to put the object in,
-        # if any
-        if "namespace" in yml_object["metadata"]:
-            namespace = yml_object["metadata"]["namespace"]
-        else:
-            namespace = "default"
-        # Expect the user to create namespaced objects more often
-        if hasattr(k8s_api, "create_namespaced_{0}".format(kind)):
-            resp = getattr(k8s_api, "create_namespaced_{0}".format(kind))(
-                body=yml_object, namespace=namespace, **kwargs)
-        else:
-            resp = getattr(k8s_api, "create_{0}".format(kind))(
-                body=yml_object, **kwargs)
-        if verbose:
-            print("{0} created. status='{1}'".format(kind, str(resp.status)))
-        return k8s_api
+        yml_document_all = yaml.safe_load_all(f)
+        # Load all documents from a single YAML file
+        for yml_document in yml_document_all:
+            # If it is a list type, will need to iterate its items
+            if "List" in yml_document["kind"]:
+                # Could be "List" or "Pod/Service/...List"
+                # This is a list type. iterate within its items
+                kind = yml_document["kind"].replace("List", "")
+                for yml_object in yml_document["items"]:
+                    # Mitigate cases when server returns a xxxList object
+                    # See kubernetes-client/python#586
+                    if kind is not "":
+                        yml_object["apiVersion"] = yml_document["apiVersion"]
+                        yml_object["kind"] = kind
+                    try:
+                        k8s_api_all.append(create_from_yaml_single_item(
+                            k8s_client, yml_object, verbose, **kwargs))
+                    except client.rest.ApiException as api_exception:
+                        print(
+                            "Error when creating {0}/{1}".format(
+                                yml_object["kind"],
+                                yml_object["metadata"]["name"]),
+                            api_exception)
+            else:
+                # This is a single object. Call the single item method
+                try:
+                    k8s_api_all.append(create_from_yaml_single_item(
+                        k8s_client, yml_document, verbose, **kwargs))
+                except client.rest.ApiException as api_exception:
```

Comment:
it is great that the util runs through the entire list even when there is a failure, like kubectl does

however kubectl returns error code 1 to stderr when failure happens, I'd expect we raise error to upstream after we finish the entire list, to make it easier for automation

Reply:
instead of `print` each failure. I'd like to see we concatenate the errors and raise in the end

for question about what the raised error should look like, as long as the error clearly identifies failed objects and is well-formatted to parse (for upstream automation) when necessary, I'm okay with having either a string or a new class for the error message. @mbohlool @yliaog Do you have suggestion on what's the best practice?

below is what kubectl does
```
$ kubectl create -f hack/testdata/list.yaml 1>/dev/null
Error from server (AlreadyExists): services "list-service-test" already exists
Error from server (AlreadyExists): deployments.apps "list-deployment-test" already exists
```